directory.nim

```diff
@@ -0,0 +1,83 @@
+## Build the contents of `directory.md` for The Algorithms/Nim repository
+##
+## Usage: `nim r directory.nim > DIRECTORY.md`
```

Comment:
We probably could write directly to DIRECTORY.md (collect every output in a string and use writeFile), since it is unlikely it will be used in other ways.

Reply:
I usually prefer when overwriting the file requires a deliberate action on user's/script's part, this achieves the goal without needing to parse arguments. Also, having the ability to pipe to a pager for preview is also nice.

Reply:
Ok, in Windows I sometime run into issues with piping and not sure if I will need to add verbosity argument to avoid having compiler output in the file.

Reply:
Hope the compiler prints to stderr, at least on `r`. :)
Well, output redirection to file never failed me on Windows, the output in the OP is from Windows too.